### PR TITLE
feat(icon-button): updates for evo design system #1491

### LINKED ISSUE #1491: icon-button: design system updates (EVO)
This ticket is specifically for the icon-btn,

Delta
Specs
Variants

### PR DESCRIPTION
**Pull request by austingardner**
PLEASE SQUASH WHEN MERGING

## Description

UPDATED: scope was reduced. See https://github.com/eBay/skin/pull/1477#issuecomment-869373805

This PR revolves around the styles for the Icon Button and Badge for Evo. It adds the badge to the Cart and Save icons, as well as an optional matte grey background. 

Ian a question for you: can we use line height in this case? It centers the number in the badge correctly; it looks really good. 

## References
<strike>part of #1433</strike> closes #1491 

## Screenshots (Mocks)
Screen Shot 2021-06-17 at 2 37 15 PM
Screen Shot 2021-06-17 at 2 37 22 PM
Screen Shot 2021-06-17 at 2 37 31 PM
Screen Shot 2021-06-17 at 2 37 37 PM
Screen Shot 2021-06-17 at 2 38 24 PM

### CODE REVIEW

**Comment by ianmcburnie** [comment]
Regarding the centering and line-height, did you test in different browsers and at different text-zoom (you need firefox for to test this) & page-zoom levels?

**Comment by austingardner** [comment]
I did not. If you think it's going to break it would probably be easier to just add a flex centering strategy.

**Comment by ianmcburnie** [comment]
I'm not sure. Best way is for you to test :-) We should be doing these type of tests for most PRs anyway.

**Comment by austingardner** [comment]
Page zoom worked but text zoom definitely broke it; Yeah I'll change this to a flex centering strategy.

**Comment by ianmcburnie** [comment]
You could also try `vertical-align`

File: src/less/icon-button/base/icon-button.less [hunk 1]
```diff
@@ -12,6 +12,35 @@ a.icon-link {
...
+button.icon-btn--save svg {
+    height: 14px;
+    width: 16px;
+}
+
+button.icon-btn--save span.badge {
+    left: 18px;
+    top: -12px;
+}
+
```
Comment (ianmcburnie) [inline src/less/icon-button/base/icon-button.less:43]: Is there a reason for all these specific styles that we didn't have before?

File: docs/_includes/common/icon-button.html [hunk 2]
```diff
@@ -54,4 +54,48 @@ <h3 id="icon-button-badged">Badged Icon Button</h3>
 </button>
     {% endhighlight %}
 
+    <h3 id="icon-button-cart-nav"> Cart Nav Icon Button </h3>
```
Comment (ianmcburnie) [inline docs/_includes/common/icon-button.html:57]: These two examples feel too specific. We just need a single example, and then clearly state that any icon can go on the grey matte background.

File: docs/_includes/common/icon-button.html [hunk 3]
```diff
@@ -54,4 +54,48 @@ <h3 id="icon-button-badged">Badged Icon Button</h3>
...
+    <p>DS Evo includes a new cart icon button with a grey background.</p>
+    <p>It requires an additional couple classes on the button - icon-btn--nav icon-btn--cart - as well as the icon-cart class on the svg.</p>
+
+    <div class="demo">
+        <div class="demo__inner">
+            <button aria-label="menu" type="button" class="icon-btn icon-btn--badged icon-btn--nav icon-btn--cart">
+                <svg class="icon icon--cart" aria-hidden="true" focusable="false" width="24" height="24"><use xlink:href="#icon-cart"></use></svg>
```
Comment (ianmcburnie) [inline docs/_includes/common/icon-button.html:64]: There seems to be a problem with the icons. Can you be sure to use the Jekyll include, e.g.:

```
{% include common/symbol.html name="settings" %}
```

Screen Shot 2021-06-21 at 6 02 13 PM

File: dist/icon-button/ds6/icon-button.css [hunk 4]
```diff
@@ -82,6 +110,7 @@ a.icon-link--badged .badge {
   display: block;
   height: 24px;
   left: 16px;
+  line-height: 16px;
```
Comment (ianmcburnie) [inline dist/icon-button/ds6/icon-button.css:113]: Thinking about it, vertically centering with line-height should be fine in this situation, so I take back what I said earlier. You might have better success though if it matches the height of the badge. You might be able to do away with the `padding-top` too.

**Comment by ianmcburnie** [comment]
I have reduced the scope of this PR to just the grey background for icon-btns. The position, size and layout of the badge should be tackled as a separate PR.

## DS6 Screenshots

Screen Shot 2021-06-27 at 10 33 23 PM

Screen Shot 2021-06-27 at 10 32 23 PM

## DS4 Screenshots

Screen Shot 2021-06-27 at 10 33 28 PM

Screen Shot 2021-06-27 at 10 32 43 PM

**Review by ianmcburnie — APPROVED** [review]
See: https://github.com/eBay/skin/pull/1477#issuecomment-869373805

**Comment by ianmcburnie** [comment]
Separate issue created for badge adjustments: https://github.com/eBay/skin/issues/1493

**Review by agliga — APPROVED** [review]

**Pull request merged**